utils/mods.py

```diff
@@ -112,6 +114,18 @@ def _get_version(self) -> str:
         return next(filter(lambda t: self.version_regex.fullmatch(t), tags))
 
 
+@dataclass(kw_only=True)
+class HexTweaksRegistryModInfo(_BaseRegistryModInfo):
+    registry_regex: re.Pattern[str] = re.compile(
+        r'PatternRegistry.mapPattern\([\n ]+HexPattern\.fromAngles\("([qweasd]+)", ?HexDir\.(.+)?\)[,\n ]+?new ResourceLocation\(".+"(.+)?"\),\n.+,(.+)'

```

Comment:
Registry regex is missing the classname. Also, the `is_great` bool group should only match if it's true — `"false"` coerces to True, which would make it treat every pattern as a great spell. See Hexal's regex for an example of both of these.

Reply:
need class name also? okay lemme quickly get that

Reply:
added classname is now capture 4